### PR TITLE
Correct property names and improve scope-names

### LINKED ISSUE #48: Scope names are inconsistent
GitHub now supports highlighting of Blade files (thanks to this package 🎉) however, it's unable to highlight the `@end` scopes due to inconsistent names.

https://github.com/github/linguist/issues/3119#issuecomment-232890748

As @Alhadis suggested, the repo needs to follow the [guidelines](http://manual.macromates.com/en/language_grammars#naming_conventions) on naming conventions.

### LINKED ISSUE #53: GitHub syntax
Hi,

Got forwarded from the GitHub linguist repository (github/linguist#3185), currently, GitHub does not highlight end calls, such as `@endif` or `@endforeach`.

Example:

```blade
@if ($errors->first())
	<div class="row">
		<div class="sixteen wide widescreen sixteen wide computer sixteen wide tablet sixteen wide mobile column">
			<div class="ui negative message">
				<div class="header">
					There's been an error.
				</div>
				<ul class="list">
					@foreach ($errors->all() as $error)
						<li>{{ $error }}</li>
					@endforeach
				</ul>
			</div>
		</div>
	</div>
@endif
```

### PR DESCRIPTION
**Pull request by Alhadis**
Fixes #48 and fixes #53.

### CODE REVIEW

**Comment by Ingramz** [comment]
Thanks, I'll check it out.

File: grammars/blade.cson [hunk 1]
```diff
@@ -11,21 +11,25 @@
...
         'name': 'comment.block.meta.embedded.blade'
       }
       {
         'begin': '(?<!@){{{'
-        'captures':
+        'beginCaptures':
           '0':
-            'name': 'support.punctuation.blade'
-          '1':
-            'name': 'source.php'
```
Comment (Ingramz) [inline grammars/blade.cson]: Leave these kind captures in, this is a hack required for text editors for properly detect inner scope when the contents of the block is 0 width, like `{{|}}` where `|` is cursor.

Comment (Alhadis) [inline grammars/blade.cson]: As a way to trigger PHP-specific snippets and commands, I assume?

Why not move the `source.php` scope to the containing block's `name` property, then...?

Comment (Alhadis) [inline grammars/blade.cson]: I've restored the capturing groups, but you should perhaps consider adding a comment to elaborate on why they're necessary. Future contributors may also misinterpret those groups as a mistake.

File: grammars/blade.cson [hunk 2]
```diff
@@ -11,21 +11,27 @@
...
+            'name': 'punctuation.section.begin.embedded.blade'
```
Comment (Ingramz) [inline grammars/blade.cson]: Move .begin and .end after .embedded like you did with comment.

File: grammars/blade.cson [hunk 3]
```diff
@@ -80,14 +92,14 @@
         ]
       }
       {
-        'match': '(?<![A-z0-9_@])@(append|break|continue|empty|else|end(can(not)?|for(each|else)?|i(f|nject)|p(hp|ush)|section|unless|verbatim|while)|overwrite|s(how|top)|verbatim)\\b|(?<!@)@parent'
-        'name': 'meta.embedded.keyword.blade'
+        'match': "(?<![A-Za-z0-9_@])@(?:append|break|continue|empty|else|end(?:can|cannot|for|foreach|forelse|if|inject|php|push|section|unless|verbatim|while)|overwrite|show|stop|verbatim)\\b|(?<!@)@parent"
```
Comment (Ingramz) [inline grammars/blade.cson]: Use single quotes

Comment (Alhadis) [inline grammars/blade.cson]: Yeah, done.

**Comment by Alhadis** [comment]
@Ingramz There's an [emergency release](https://github.com/github/linguist/issues/3204#issuecomment-246552378) of Linguist due to be cut pretty soon, and it'd be nice to get these highlighting issues fixed on the site instead of wait a few more weeks. :)

**Comment by Ingramz** [comment]
@Alhadis alright. It'll have to wait till I am done with my work commitments for the day, though.

**Comment by Ingramz** [comment]
@Alhadis I'm sorry it didn't make the cut (pun intended). But here's my offer regarding scope names:
https://docs.google.com/spreadsheets/d/1bTwlO-cSer1rI4lY-IptddjYUQq4dkJiudM1gTB1BgU/edit?usp=sharing

Bold marks differences. Let me know what you think. I think it's a slightly better compromise between properly defined names and not looking like crap, but definitely not perfect. I will probably start splitting built-in directives into smaller groups by their function, eg control keywords and other keywords, but let that be for now.

Is there any way to test this configuration against linguist ahead of time?

**Comment by Alhadis** [comment]
> Is there any way to test this configuration against linguist ahead of time?

Well, there's a webapp called [Lightshow](https://github-lightshow.herokuapp.com/) which will give a general impression, but its highlighting palette is outdated and the app itself is buggy. Should be enough though.

**Comment by Ingramz** [comment]
At least with lightshow the echo mustaches look pretty now 😸

**Comment by Alhadis** [comment]
Well, at the end of the day, that's all that matters, I guess. :)

Pretty moustaches an' shit.

File: grammars/blade.cson [hunk 4]
```diff
@@ -51,72 +60,75 @@
...
           '0':
-            'name': 'meta.embedded.keyword.blade'
+            'name': 'keyword.begin.blade'
         'endCaptures':
           '0':
-            'name': 'meta.embedded.keyword.blade'
+            'name': 'keyword.end.blade'
         'contentName': 'source.php'
-        'name': 'meta.embedded.blade'
+        'name': 'meta.embedded.keyword.blade'
```
Comment (Ingramz) [inline grammars/blade.cson]: should be `meta.embedded.block.blade`

Comment (Alhadis) [inline grammars/blade.cson]: Remedied.

**Comment by Ingramz** [comment]
Sweet. Expect a release very soon™.

**Comment by Alhadis** [comment]
Ace, now people will shaddap at Linguist. 👍

**Comment by Alhadis** [comment]
Aitäh!

**Pull request merged**